### PR TITLE
gh-123430: Add dark mode support to pages generated by http.server

### PR DESCRIPTION
**Pull request by YorikHansen**
The error pages and the directory lists generated by http.server did not allow browsers to enable the inbuilt dark mode. This change allows for that by setting the css property color-scheme properly.

* Issue: gh-123430

### CODE REVIEW

File: Misc/NEWS.d/next/Library/2024-08-29-14-51-36.gh-issue-123430.M7wXl9.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Pages generated by the ``http.server`` module allow the brower to apply it's default dark mode.
```
Comment (ZeroIntensity) [inline Misc/NEWS.d/next/Library/2024-08-29-14-51-36.gh-issue-123430.M7wXl9.rst]: Nitpick:
```suggestion
Pages generated by the :mod:`http.server` module allow the brower to apply its default dark mode.
```

Comment (YorikHansen) [reply Misc/NEWS.d/next/Library/2024-08-29-14-51-36.gh-issue-123430.M7wXl9.rst]: Thank you :-)

**Review by hugovk — APPROVED** [review]
## `main`

Testing `./python.exe -m http.server` and looking at a directory listing and an error page:

image

image


It's the same in light and dark mode on Chrome with macOS.

## PR

In dark mode:

image

image

And in light mode it's the same as `main`.

File: Misc/NEWS.d/next/Library/2024-08-29-14-51-36.gh-issue-123430.M7wXl9.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Pages generated by the :mod:`http.server` module allow the brower to apply its default dark mode.
```
Comment (vstinner) [inline Misc/NEWS.d/next/Library/2024-08-29-14-51-36.gh-issue-123430.M7wXl9.rst]:
```suggestion
Pages generated by the :mod:`http.server` module allow the browser to apply its default dark mode.
```

**Review by vstinner — APPROVED** [review]
I cannot review the CSS, but I like @hugovk's screenshots and the feature.

**Comment by ZeroIntensity** [comment]
Should this get backported? The user seeing the incorrect color scheme sounds like a bug to me.

**Comment by vstinner** [comment]
> Should this get backported? The user seeing the incorrect color scheme sounds like a bug to me.

It's a new feature, it should not be backported.

**Comment by hugovk** [comment]
@YorikHansen Please could you also mention this in What's New?

https://docs.python.org/3.14/whatsnew/3.14.html#improved-modules

**Comment by hugovk** [comment]
Thank you!

**Pull request merged**